pkg/object/meshcontroller/spec/spec.go

```diff
@@ -312,6 +313,39 @@ func (cr CustomResource) Kind() string {
 	return ""
 }
 
+// MarshalJSON implements json.Marshaler
```

Comment:
You can do a research of https://github.com/json-iterator/go , which might already have done it.

Reply:
yes, `json-iterator` can handle this case, thanks for the advice.

btw, I think the document of `json-iterator` is inaccurate as it says "100% compatible with `encoding/json`", at least in this case.

Reply:
Compatible means it includes standard lib here. it depends on how you understand it. Golang1.17 is compatible with Golang 1.0 with bringing a lot of new features.

Reply:
After thinking the issue twice, I think it is better to keep this function: 

This function is always a correct solution, no matter we are using `encoding/json` or `json-iterator`. 
If we remove it, then we are forcing future code to use `json-iterator`, as this requirement is an implicit one, we may meet this bug again if `encoding/json` is used in new code.

Reply:
Sure, I have no opposite opinion

Reply:
Because I found other issues with this implementation, I've moved the type conversion stuff to `UnmarshalYAML`.